indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -581,6 +649,9 @@ protected void reduce(
             toMerge.add(finalFile);
           }
 
+          Futures.allAsList(persistFutures).get(1, TimeUnit.HOURS);
```

Comment:
on second thought, it is probably ok as 1 hr should be practically infinite in this case.